apps/rand.c

```diff
@@ -95,8 +97,51 @@ int rand_main(int argc, char **argv)
     argc = opt_num_rest();
     argv = opt_rest();
     if (argc == 1) {
+        int factoridx = 0;
+        int scale = 0;
+        /*
+         * iterate over the value and check to see if there are
+         * any non-numerical chars
+         * A non digit suffix indicates we need to scale the
+         * number of requested bytes by a factor of:
+         * k = 1024^1 (1 << (10 * 1))
+         * m = 1024^2 (1 << (10 * 2))
+         * g = 1024^3 (1 << (10 * 3))
+         */
+        while (argv[0][factoridx]) {
+            if (!isdigit((int)(argv[0][factoridx]))) {
+                switch(argv[0][factoridx]) {
+                case 'k':
+                    scale = 10;
+                    break;
+                case 'm':
+                    scale = 20;
+                    break;
+                case 'g':
+                    scale = 30;
+                    break;
+                default:
+                    goto opthelp; 
+                }
+                break;
+            }
+            factoridx++;
+        }
+        /* Remove the suffic from the arg so that opt_long works */
```

Comment:
Typo: `suffix`

And: do you want to check that there's nothing after the scaling character, or do we not care? (i.e. how strict about being something like 10m vs 10mb vs 10mejcdufh?). Also should we allow `10M` as an alternative?

Reply:
I thought it would be ok to allow trailing garbage, as it just wouldn't be considered, but if you feel like we need to be strict about it, I can add a check for that